README.md

```diff
@@ -230,29 +308,128 @@ MQTT over TLS/SSL | dashboard.example.com | 8883 | Username/Password come from m
+
+This is the gateway for adding devices to the DNC Server. This record contains all devices entries irrespective of Clients.
+
+| Fields | Description |
+|--------|-------------|
+| Client | Name Device will be assigned to this Client |
+|Hardware ID | ID of the Hardware printed in the PCB |
+| deviceID | ID received from TTN/Sigfox – This data should be existed as a tag in the influxDB database |
+| devID | ID received from TTN/Sigfox– This data should be existed as a tag in the influxDB database |
+| devEUI | ID received from TTN/Sigfox– This data should be existed as a tag in the influxDB database |
+| Measurement Name | Where the device data gets logged in the Database Server. (In InfluxDB it is called as measurement name)|
+| Field Data Name | Name of the Data which required by the Client.  Example [ tWater/rh/vBat etc.] This data should be existed as a field tag in the influxDB database |
+| Date of Installation | The date when the device installed in the field |
+| Data of Removal | The date when device removed from the field. This will not be asked while adding device, required only remove/replace device |
+
+The Master Admin has the access to manage this record, no other can do any changes in this record.
+**Note:-** The deviceID, devID and devEUI are optional but any-one should be mandatory.
+
+##### Devices
+
+This is the gateway for adding devices under a client with the customized tag details. All tag details are optional, the admin can add device with or without tag fields.
+
+| Fields | Description |
+|------- | ----------- |
+| Hardware ID | Select from the Master Record, direct entry not allowed |
+| Tag 1 - optional | Value for the Tag |
+| Tag 2 - optional | Value for the Tag N - optional Value for the Tag |
+| Latitude | Location coordinates – Required when showing data on world map |
+| Longitude | Location coordinates – Required when showing data on world map |
+| Installation Date | Get it from the Master record |
+| Removal Date | Enter the date when the device removed from the field. This date automatically updated to Master record. |
+
+#### Plugins
+
+Plugins are created to provide User interface for using DNC application, which communicates with DNC engine by using the Plugin API.
+
+##### Standard Plugin API
+
+This API receives the request from Excel sheet or Google sheet Plugin, communicates with the DNC server, InfluxDB and sends back the response to the Plugin.
+
+##### Grafana Influx Plugin API
+
+This API receives requests from Grafana application, which are influxDB based queries. In Grafana UI all the requests are redirected to Grafana Influx Plugin, which communicates with DNC server, InfluxDB and then sends the response back to Grafana UI.
+
+### DNC server Architecture
+
+![dnc_server_arch](assets/dnc_server_arch.png)
+
 ## Setup Instructions
 
-Please refer to [`SETUP.md`](./SETUP.md) for detailed set-up instructions.
+Please refer to [`SETUP.md`](./SETUP.md) for detailed basic set-up instructions.
```

Comment:
the word "basic" is redundant with (and contradicts) "detailed". Use one word or the other.

Reply:
Fixed it.